src/humanize/filesize.py

```diff
@@ -83,11 +85,7 @@ def naturalsize(
         suffix = suffixes["decimal"]
 
     base = 1024 if (gnu or binary) else 1000
-    if isinstance(value, str):
-        bytes_ = float(value)
-    else:
-        bytes_ = value
-
+    bytes_ = float(value)
```

Comment:
It's also faster to simply cast to float rather than do `isinstance` call.

Reply:
Yep:


```
❯ python --version
Python 3.14.0b1

❯ python -m timeit -s "value = 123.456" "bytes_ = float(value) if isinstance(value, str) else value"
10000000 loops, best of 5: 24.9 nsec per loop

❯ python -m timeit -s "value = 123.456" "bytes_ = float(value)"
20000000 loops, best of 5: 13.7 nsec per loop

❯ python -m timeit -s "value = '123.456'" "bytes_ = float(value) if isinstance(value, str) else value"
5000000 loops, best of 5: 47.7 nsec per loop

❯ python -m timeit -s "value = '123.456'" "bytes_ = float(value)"
10000000 loops, best of 5: 36.2 nsec per loop
```

Although in isolation this change would fail:

```
AssertionError: assert '1000.0 ZB' == '1.0 YB'
```